src/PeNet/Header/Resource/Icon.cs

```diff
@@ -56,44 +56,42 @@ public Span<byte> AsRawSpan()
             if (IsPng) return raw.ToArray(); // TODO: CFF does not use an additional header for .PNG. But this would not break anything.
             if (!IsIco) return null;
 
-            var icoBytes = new byte[IcoHeaderSize + raw.Length];
-            var icoHeader = IcoHeader();
+            var header = WriteIconDirectoryStructure();
+            var directory = WriteIconDirectoryEntryStructure();
+            var icoBytes = new byte[header.Length + directory.Length + raw.Length];
 
-            icoHeader.AsSpan().CopyTo(icoBytes, 0);
-            raw.CopyTo(icoBytes, (int)IcoHeaderSize);
+            header.AsSpan().CopyTo(icoBytes, 0);
+            directory.AsSpan().CopyTo(icoBytes, header.Length);
+            raw.CopyTo(icoBytes, header.Length + directory.Length);
             return icoBytes;
         }
 
-        private byte[] IcoHeader()
+        private static byte[] WriteIconDirectoryStructure()
         {
-            var stream = new MemoryStream((int) IcoHeaderSize);
-            WriteIconDirectoryStructure(stream);
-            WriteIconDirectoryEntryStructure(stream);
-            return stream.ToArray();
+            var header = new byte[IcoHeaderSize];
+            ((ushort) 0).LittleEndianBytes().AsSpan().CopyTo(header, 0);
+            ((ushort) 1).LittleEndianBytes().AsSpan().CopyTo(header, 2);
+            ((ushort) 1).LittleEndianBytes().AsSpan().CopyTo(header, 4);
+            return header;
         }
 
-        private static void WriteIconDirectoryStructure(Stream stream)
+        private byte[] WriteIconDirectoryEntryStructure()
         {
-            stream.Write(((ushort) 0).LittleEndianBytes()); // Res
-            stream.Write(((ushort) 1).LittleEndianBytes()); // Type .ICO
-            stream.Write(((ushort) 1).LittleEndianBytes()); // Num of icons in file (=1)
-        }
-
-        private void WriteIconDirectoryEntryStructure(Stream stream)
-        {
-            stream.WriteByte(AssociatedGroupIconDirectoryEntry!.BWidth); //Width
-            stream.WriteByte(AssociatedGroupIconDirectoryEntry!.BHeight); //Height
+            var directory = new byte[IcoDirectorySize];
+            directory[0] = AssociatedGroupIconDirectoryEntry!.BWidth; //Width
+            directory[1] = AssociatedGroupIconDirectoryEntry!.BHeight; //Height
 
             //Information not included in the GroupIconDirectoryEntry, only in the image byte array for .BMP. By default 0x00.
-            stream.WriteByte(AsRawSpan()[32]); //Number of Colors in color palette
-            stream.WriteByte(0x00); //Res
+            directory[2] = AsRawSpan()[32]; //Number of Colors in color palette
```

Comment:
Nit: Adapting to format of the others?:
```
AsRawSpan()[32].CopyTo(directory, 2);
new byte[] {0x00}.CopyTo(directory, 3);
```

Reply:
I don't prefer one over the other, so I'm fine with both